shared/src/main/scala/coop/rchain/shared/Log.scala

```diff
@@ -11,6 +17,21 @@ object LogSource {
   def apply(c: Class[_]): LogSource = new LogSource {
     val clazz: Class[_] = c
   }
+
+  implicit def matLogSource: LogSource = macro LogSourceMacros.mkLogSource
```

Comment:
can you provide context what this macro is doing? maybe it's worth documenting in comments?

Reply:
I mean, I assume it adds automatically `LogSource` but I don't see usage so I'm not 100% certain

Reply:
It is used in `GenerateCertificateIfAbsent` but I think we could eventually replace all manual definitions.

Reply:
awesome